zeebe/exporter-common/src/main/java/io/camunda/zeebe/exporter/common/auditlog/AuditLogInfo.java

```diff
@@ -169,6 +172,21 @@ private static AuditLogOperationType getOperationType(final Intent intent) {
       case UserIntent.DELETED:
         return AuditLogOperationType.DELETE;
 
+      case UserTaskIntent.ASSIGNED:
+        // Operation type is UNASSIGN if assignee is not provided
+        final String assignee = ((UserTaskRecordValue) record.getValue()).getAssignee();
+        return (assignee == null || assignee.isBlank())
+            ? AuditLogOperationType.UNASSIGN
+            : AuditLogOperationType.ASSIGN;
```

Comment:
🔧 Let's add a guard clause here to assert that the record is `instanceof UserTaskRecordValue`, just in case we ever introduce a new record value that maps `UserTaskIntent.ASSIGNED`.

If the guard condition fails, we should return a `AuditLogOperationType.UNKNOWN`.